2-mandatory/3-financial-times.js

```diff
@@ -23,59 +34,74 @@ function titleWithFewestWords(allArticleTitles) {
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+  return allArticleTitles.filter((title, index, array) => {
+    const matches = title.match(/[0123456789]/); // will be an array (if have matches) or null (if we have no matches)
+    if (matches !== null) {
+      return true;
+    } else {
+      return false;
+    }
+  });
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
```

Comment:
A version to think about - have a read about `map` and `reduce` - these are quite advanced concepts, that may be very useful in practice. Try to understand how the following code works. You may even decide that it is beautiful :)

```javascript
let average = allArticleTitles.map(a => a.length).reduce((a, b) => a + b) / allArticleTitles.length
return Math.round(average)
```